pepper-apis/studybuilder-cli/tenants/atcp/actions/google-signin-restriction.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * This Action was migrated from Rule.
+ * Rule name: Google Sign In Restriction
+ * Rule ID: rul_
+ * Created on 9/24/2024
+ */
+
+/**
+ * Handler that will be called during the execution of a PostLogin flow.
+ *
+ * @param {Event} event - Details about the user and the context in which they are logging in.
+ * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.
+ */
+exports.onExecutePostLogin = async (event, api) => {
+  /**
+   * The following code block will skip this Action if Rule "Google Sign In Restriction"
+   * was previously executed in the transaction in order to avoid duplication
+   * of logic.
+   */
+  if (api.rules.wasExecuted("change_me")) {
+    return;
+  }
+
+  // YOUR_CODE_HERE
+  const GOOGLE_CONNECTION_IDENTIFIER = 'google-oauth2';
+
+  const app_metadata = event.user.app_metadata || {};
+  const pepperUserGuid = event.user.app_metadata.user_guid;
+  const connectionType = event.connection.name;
+  console.log('Connection: ' + JSON.stringify(event.connection));
+
+  if (connectionType === GOOGLE_CONNECTION_IDENTIFIER) {
+    if (pepperUserGuid) {
+      console.log('All good, user has pepper_user_guid: ' + pepperUserGuid);
+      return;
+    } else {
+      /**
+       * User was not registered in pepper yet
+       */
+      console.log('Nooo, there is no pepper user guid...', { pepperUserGuid });
+      console.log('User', JSON.stringify(event.user));
+      console.log('Event', JSON.stringify(event));
+
+      const request = event.request || {};
+      const query = request.query || {};
+
+      const tempUserGuid = query.temp_user_guid;
+
+      if (!tempUserGuid) {
+        console.log('There is no temp user guid...', { tempUserGuid });
+
+        const loginErrPayload = {
+          code: 'prequal_skipped',
+          message: 'You have to complete prequal first in order to proceed',
```

Comment:
Where do these error messages show up for the user?  I'm wondering if our users will know what "prequal" means.

Reply:
Some cases front-end is displaying to users but In this case, I dont think front-end is displaying to user. Looks like forwarding to prequal page. will double check with updated code. Only AT has it.